horace_core/algorithms/upgrade_legacy_alignment.m

```diff
@@ -0,0 +1,76 @@
+function out = upgrade_legacy_alignment(in_data,alatt0,angdeg0)
+% Change the crystal lattice and orientation of an sqw object stored in a file
+% or celarray of files
+%
+%   >> upgrade_legacy_alignment(filenames, alatt0,angdeg0)
+% modify legacy alignment previously
+%                                                 % of the crystal according to the
+%                                                 % crystal alignment information provided
+%
+%
+% Input:
+% -----
+%   w          cellarray of Input sqw objects or file names for sqw files,
+%              containing sqw objects, aligned using legacy alignment
+%
+%  alatt0   -- vector of lattice parameters for initial not aligned file
+%  angdeg0  -- vector of lattice angles for initial not aligned file.
+%
+
+% Output:
+% -------
+%   wout      cellarray of iutput sqw objects and filename containing sqw
+%             objects modified
+%
+
+% This routine used to change the crystal in sqw files, when it overwrites the input file.
+
+% Parse input
+% -----------
+if ischar(in_data) % process single file
+    in_data = {in_data};
+end
+if ~isnumeric(alatt0) || ~isnumeric(angdeg0) || numel(alatt0) ~= 3 || numel(angdeg0) ~= 3 || nargin < 3
```

Comment:
Do we want to make `is_3vector` and `is_3x3matrix` from `Symop` a public utility function?

Reply:
I am not sure but probably it would rather complicate the code understanding then simplify it. Yes, experienced users would benifit from it, but you will forget it after some time. 
The code here is not very complicated and easy to read (and I have some similar checks in projection too) so I would leave it as it is. But not sure.

Reply:
That's kind of the point. We have a lot of points where we check for 3-vectors and 3x3 matrices. 
```
if ~is_3vector(alatt0) || ~is_3vector(angdeg) || nargin < 3
```
Isn't exactly compicated